packages/pros/src/io/print_impl.rs

```diff
@@ -0,0 +1,318 @@
+//! Implements `println!`, `eprintln!` and `dbg!` on top of the `pros_sys` crate without requiring
+//! the use of an allocator. (Modified version of `libc_print` crate)
+//!
+//! Allows you to use these macros in a #!\[no_std\] context, or in a situation where the
+//! traditional Rust streams might not be available (ie: at process shutdown time).
+//!
+//! [`writeln`] and [`ewriteln`] are provided for cases where you may not wish
+//! to pull in the overhead of the formatter code and simply wish to print C-style strings.
+//!
+//! ## Usage
+//!
+//! Exactly as you'd use `println!`, `eprintln!` and `dbg!`.
+//!
+//! ```rust
+//! # use pros::io::print_impl::*;
+//! // Use the default ``-prefixed macros:
+//! # fn test1()
+//! # {
+//! println!("Hello {}!", "stdout");
+//! eprintln!("Hello {}!", "stderr");
+//! let a = 2;
+//! let b = dbg!(a * 2) + 1;
+//! assert_eq!(b, 5);
+//! # }
+//! ```
+//!
+//! Or you can import aliases to `std` names:
+//!
+//! ```rust
+//! use pros::io::print_impl::{println, eprintln, dbg};
+//!
+//! # fn test2()
+//! # {
+//! println!("Hello {}!", "stdout");
+//! eprintln!("Hello {}!", "stderr");
+//! let a = 2;
+//! let b = dbg!(a * 2) + 1;
+//! assert_eq!(b, 5);
+//! # }
+//! ```
+
+// libc_print is licensed under the MIT License:
+
+// Copyright (c) 2023 Matt Mastracci and contributors
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#![no_std]
+#![allow(dead_code)]
+#![allow(unused)]
+#![warn(unsafe_op_in_unsafe_fn)]
+
+use core::{convert::TryFrom, file, line, stringify};
+
+/// This forces a "C" library linkage
+#[cfg(not(windows))]
+#[link(name = "c")]
+mod c {
+    extern "C" {}
+}
```

Comment:
why is this necessary?

Reply:
I don't think its correct here because we don't actually have to link with libc here. These come from libpros.